pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -180,3 +184,85 @@ type HTTPSDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// HTTPSDConfig defines a prometheus HTTP service discovery configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#consul_sd_config
+// +k8s:openapi-gen=true
+type ConsulSDConfig struct {
+	// A valid string consisting of a hostname or IP followed by an optional port number.
+	// +kubebuilder:validation:MinLength:=1
+	// +required
+	Server string `json:"server,omitempty"`
+	// Consul ACL Token, If not provided it will use the ACL from the local Consul Agent.
+	// +optional
+	Token *corev1.SecretKeySelector `json:"token,omitempty"`
```

Comment:
(suggestion) a Ref suffix makes it more obvious that it's a reference to an external object, not a scalar value.
```suggestion
	TokenRef *corev1.SecretKeySelector `json:"tokenRef,omitempty"`
```